nncf/experimental/torch/fx/quantization/quantize_pt2e.py

```diff
@@ -0,0 +1,151 @@
+    subset_size: int = 300,
+    fast_bias_correction: Optional[bool] = True,
+    smooth_quant: bool = False,
+    bias_correction_params: Optional[AdvancedBiasCorrectionParameters] = None,
+    smooth_quant_params: Optional[AdvancedSmoothQuantParameters] = None,
+    activations_range_estimator_params: Optional[RangeEstimatorParameters] = None,
+    weights_range_estimator_params: Optional[RangeEstimatorParameters] = None,
+    batchwise_statistics: bool = False,
+    fold_quantize: bool = False,
+) -> torch.fx.GraphModule:
+    """
+    Applies post-training quantization to the torch.fx.GraphModule provided model
+    using provided torch.ao quantizer.
+
+    :param quantizer: Torch ao quantizer to annotate nodes in the graph with quantization setups
+        to convey the desired way of quantization.
+    :param subset_size: Size of a subset to calculate activations
+        statistics used for quantization.
+    :param fast_bias_correction: Setting this option to `False` enables a different
+        bias correction method which is more accurate, in general, and takes
+        more time but requires less memory. None disables the bias correction algorithm.
+    :param smooth_quant: Setting this option to `True` enables the SmoothQuant algorithm.
+    :param bias_correction_params: Contains advanced parameters for fine-tuning bias correction algorithm.
+    :param smooth_quant_params: Contains advanced alpha parameters for SmoothQuant algorithm.
+    :param activations_range_estimator_params: Contains parameters for estimating the range
+        of activations of the model.
+    :param weights_range_estimator_params: Contains parameters for estimating the range
+        of weights of the model.
+    :param batchwise_statistics: Determines whether quantizer statistics should be calculated
+        for each item of the batch or for the entire batch, default is False.
+    :param fold_quantize: Boolean flag for whether fold the quantize op or not.
+    """
+    nncf_logger.warning(
+        "Experimental Torch FX quantization backend is being used for the given torch.fx.GraphModule model."
+        " Torch FX PTQ is an experimental feature, consider using Torch or OpenVino PTQ backends"
+        " in case of errors or a poor model performance."
+    )
+
+    if subset_size < 1:
+        raise nncf.ValidationError("Subset size must be positive.")
+
+    batch_size = calibration_dataset.get_batch_size()
+    if batch_size is not None and batch_size > 1 and batchwise_statistics is not None:
+        batchwise_statistics = True
+
+    original_graph_meta = model.meta
+
+    copied_model = deepcopy(model)
+
+    # To make it easier for bias correction algorithms,
+    # biases are being separated by the followng calls.
+    fuse_conv_bn(copied_model)
+    # Call ao quantizer transform_for_annotation
+    # before the NNCFGraph creation
+    quantizer.transform_for_annotation(copied_model)
+
+    if not isinstance(quantizer, NNCFQuantizer):
```

Comment:
I believe it is more logical to check that it is `Quantizer`,  before creating `NNCFFXQuantizer` quantizer.